proto/redpanda/api/dataplane/v1/mcp.proto

```diff
@@ -0,0 +1,664 @@
+
+  // Tags are key-value pairs that can be assigned to a MCP server resource.
+  // They help organize MCP servers and enable filtering when listing them.
+  map<string, string> tags = 9 [(buf.validate.field).map = {
+    max_pairs: 16
+    values: {
+      string: {pattern: "^([\\p{L}\\p{Z}\\p{N}_.:/=+\\-@]*)$"}
+    }
+  }];
+
+  message Resources {
+    // `memory_shares` is a string specifying the amount of memory to allocate for
+    // the MCP server.
+    //
+    // This follows the [Kubernetes quantity](https://kubernetes.io/docs/reference/kubernetes-api/common-definitions/quantity/) format. Acceptable units
+    // include:
+    // - Decimal SI units: "K", "M", "G", "T", "P", "E" (e.g., "128M" for 128
+    //   megabytes)
+    // - Binary SI units: "Ki", "Mi", "Gi", "Ti", "Pi", "Ei" (e.g., "512Mi" for
+    // 512 mebibytes) If you don't specify a unit, the value is interpreted as
+    // bytes.
+    string memory_shares = 1 [
+      (google.api.field_behavior) = REQUIRED,
+      (buf.validate.field).required = true
+    ];
+    // `cpu_shares` is a string specifying the amount of CPU to allocate for the
+    // MCP server.
+    //
+    // This follows the [Kubernetes quantity](https://kubernetes.io/docs/reference/kubernetes-api/common-definitions/quantity/) format. Acceptable
+    // units include:
+    // - Decimal SI units: "m" (e.g., "500m" for 500 millicores, "2" for 2 cores)
+    // CPU shares can be specified in millicores (1 core = 1000 millicores).
+    // If you don't specify a unit, the value is interpreted as the number of cores.
+    string cpu_shares = 2 [
+      (google.api.field_behavior) = REQUIRED,
+      (buf.validate.field).required = true
+    ];
+  }
+
+  // State of the MCP server.
+  enum State {
+    STATE_UNSPECIFIED = 0;
+    // The MCP server is starting.
+    STATE_STARTING = 1;
+    // The MCP server is running.
+    STATE_RUNNING = 2;
+    // The MCP server is in the process of stopping.
+    STATE_STOPPING = 3;
+    // The MCP server is stopped and in paused state.
+    STATE_STOPPED = 4;
+    // The MCP server encountered an error.
+    STATE_ERROR = 5;
+    // The MCP server has completed the job successfully.
+    STATE_COMPLETED = 6;
+  }
+
+  // MCP server status may contain an error message.
+  message Status {
+    string error = 2;
+  }
+}
+
+// MCPServer is the service for Redpanda Connect MCP Servers.
+// It exposes the API for creating and managing Redpanda Connect MCP servers and their configurations.
+service MCPServerService {
+  option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_tag) = {
+    name: "Redpanda Connect MCP servers"
+    description: "Create and manage Redpanda Connect MCP servers and their configurations."
+  };
+
+  // CreateMCPServer creates a Redpanda Connect MCP Server in the Redpanda cluster.
+  rpc CreateMCPServer(CreateMCPServerRequest) returns (CreateMCPServerResponse) {
+    option (google.api.http) = {
+      post: "/v1/redpanda-connect/mcp-servers"
```

Comment:
[question]: shall we keep putting this stuff under redpanda-connect? what is the benefit? should it be a completely separate resource? eg. /v1/mcp-servers

Reply:
Good question, no strong opinion on either, but I have a slight preference to group it under redpanda-connect I believe. Purely because I think it's a more specific resource type of redpanda connect and doesn't seem important/distinct enough for me to entertain it's own resources.